### ISSUE TITLE
Error producing XML with certain characters in InlineData

### ISSUE THREAD

**Issue by fedotovalex** [xunit/xunit#865]
Source code:

```
using System;
using Xunit;

namespace Test
{
    public class Test
    {
        [Theory]
        [InlineData("\b")]
        public void InvalidChar(
            string s
            )
        {
            Assert.True(Char.IsControl(s[0]));
        }
    }
}
```

project.json:
```
{
  "version": "1.0.0-*",
  "testRunner": "xunit",
  "dependencies": {
    "dotnet-test-xunit": "1.0.0-*",
    "xunit": "2.2.0-beta1-build3239"
  },
  "frameworks": {
    "net46": {}
  }
}
```

Running tests fails to produce XML output due to an invalid XML character:
```
C:\...\Test>dotnet test -xml obj\test.xml
Project Test (.NETFramework,Version=v4.6) was previously compiled. Skipping compilation.
xUnit.net .NET CLI test runner (64-bit Desktop .NET win10-x64)
  Discovering: Test
  Discovered:  Test
  Starting:    Test
  Finished:    Test
=== TEST EXECUTION SUMMARY ===
   Test  Total: 1, Errors: 0, Failed: 0, Skipped: 0, Time: 0.147s
error: ', hexadecimal value 0x08, is an invalid character.
SUMMARY: Total: 1 targets, Passed: 0, Failed: 1.
```

**Comment by fedotovalex** [comment]
Looks like this should be fixed by #807. Please...

**Comment by fedotovalex** [comment]
Confirmed fixed in v2.2.0-beta2-build3284. Thanks!
